### PR TITLE
Code refactoring and JSDocs

### PR DESCRIPTION
**Pull request by tormozz48**
@tadatuta

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,59 +1,80 @@
...
+const _ = require('lodash');
+const q = require('q');
 const rc = require('./lib/rc');
 const mergeWith = require('lodash/mergeWith');
 const glob = require('glob');
 
+/**
+ * Constructor
+ * @param {Object} [options] object
+ * @param {String} [options.name] - config filename. ('bem' by default)
```
Comment (tadatuta) [inline index.js]: [options.name=bem]

File: index.js [hunk 2]
```diff
@@ -1,59 +1,80 @@
...
+ * @param {String} [options.argv] custom path to config on FS via command line argument `--config` (defaults to `null`)
+ * @constructor
+ */
 function BemConfig(options) {
     this._options = options || {};
     this._options.projectRoot || (this._options.projectRoot = process.cwd());
 }
 
+/**
+ * Returns all of found configs
```
Comment (tadatuta) [inline index.js]: ~~of~~

File: index.js [hunk 3]
```diff
@@ -1,59 +1,80 @@
...
+ * @returns {Promise}
+ */
 BemConfig.prototype.configs = function() {
     const options = this._options;
     const projectRoot = options.projectRoot;
 
-    return Promise.all(rc({
+    const globPromise = q.denodeify(glob);
```
Comment (tadatuta) [inline index.js]: get rid of `Q`

File: index.js [hunk 4]
```diff
@@ -84,47 +105,47 @@ BemConfig.prototype.level = function(levelPath) {
...
 BemConfig.prototype.library = function(libName) {
-    return this.get().then(config => {
-        return new BemConfig({ projectRoot: config.libs[libName].path });
-    });
+    return this.get()
+        .then(config => new BemConfig({ projectRoot: config.libs[libName].path }));
 };
 
+/**
+ * Returns map of settings for each of levels
```
Comment (tadatuta) [inline index.js]: of levels -> level

File: index.js [hunk 5]
```diff
@@ -84,47 +105,47 @@ BemConfig.prototype.level = function(levelPath) {
...
+ * @returns {Promise}
+ */
 BemConfig.prototype.module = function(moduleName) {
-    return this.get().then(config => {
-        const modules = config.modules;
-
-        return modules && modules[moduleName];
-    });
-}
+    return this.get()
```
Comment (tadatuta) [inline index.js]: let's rollback

File: package.json [hunk 6]
```diff
@@ -32,9 +32,11 @@
   "homepage": "https://github.com/bem-contrib/bem-config#readme",
   "dependencies": {
     "glob": "^6.0.4",
+    "lodash": "^4.6.1",
     "lodash.mergewith": "^4.3.3",
```
Comment (tadatuta) [inline package.json]: refactor to use `lodash` as well

**Comment by tormozz48** [comment]
@tadatuta Fix notations

**Pull request merged**